coops-ui/src/components/Common/OfficeAddresses.vue

```diff
@@ -0,0 +1,813 @@
+      }
+
+      this.showAddressForm = false
+      this.emitAddresses()
+      this.emitModified()
+    }
+
+    /**
+     * Resets the address data to what it was before any edits were done.
+     */
+    private resetAddress (): void {
+      this.deliveryAddress = { ...this.deliveryAddressOriginal }
+      this.mailingAddress = { ...this.mailingAddressOriginal }
+      this.recDeliveryAddress = { ...this.recDeliveryAddressOriginal }
+      this.recMailingAddress = { ...this.recMailingAddressOriginal }
+      this.emitAddresses()
+      this.emitModified()
+    }
+    /**
+     * Compares two address objects while omitting specified properties from the comparison.
+     *
+     * @param addressA The first address to compare
+     * @param addressB The second address to compare
+     * @param prop The property to omit during the comparison
+     *
+     * @return boolean A boolean indicating a match of addresses
+     */
+    private isSameWithoutProp (addressA: AddressIF, addressB: AddressIF, ...prop: Array<string>): boolean {
+      return this.isSameAddress({ ...this.omitProp(addressA, [...prop]) }, { ...this.omitProp(addressB, [...prop]) })
+    }
+
+    /**
+     * Add an action, if it doesn't already exist; ensures no multiples.
+     */
+    private addAction (address: AddressIF, val: string): void {
+      if (address.actions.indexOf(val) < 0) address.actions.push(val)
+    }
+
+    /**
+     * Remove an action, if it already exists.
+     */
+    private removeAction (address: AddressIF, val: string): void {
+      address.actions = address.actions.filter(el => el !== val)
+    }
+
+    /**
+     * Initialize and assign base states of Address Data
+     *
+     */
+    private initAddresses (): void {
+      // If loading loading from a draft filing
+      if (this.addresses) {
+        this.addresses.registeredOffice && this.assignRegisteredAddresses(this.addresses.registeredOffice)
+        this.addresses.recordsOffice && this.assignRecordAddresses(this.addresses.recordsOffice)
+      } else {
+        // If new filing load from addresses from store
+        this.registeredAddress && this.assignRegisteredAddresses(this.registeredAddress)
+        this.recordsAddress && this.assignRecordAddresses(this.recordsAddress)
+      }
+
+      // emit address data back up so that parent data has data (needed for AR filing specifically)
+      this.emitAddresses()
+    }
+
+    /**
+     * Assign the Registered address objects actions and omit the addressType
+     *
+     * @param addressBase The base address object to be parsed and assigned to its Office Address class property
+     */
+    private assignRegisteredAddresses (addressBase: BaseAddressObjIF): void {
+    // Assign Delivery Address
+      const deliveryAddress = addressBase.deliveryAddress
+      if (deliveryAddress) {
+        deliveryAddress.actions = []
+        this.deliveryAddressOriginal = { ...this.omitProp(deliveryAddress, ['addressType']) as AddressIF }
```

Comment:
Should prop be omitted earlier, like wherever addressBase is fetched?

Also in assignRecordAddresses() below.

Reply:
Yes!  I implemented the omitProps at the app level. EntityType will no longer be based into the children components.

Reply:
Yay, thanks! (Original) business address is only fetched in 1 place now!

@cameron-freshworks Can you confirm `addressType` is not returned anywhere else (eg, director data)? I don't see it in Dev but you may have different BE code than me.

Reply:
I cannot find `addressType` locally past the point it's filtered out ( `app.vue`).

Reply:
Perfect, thanks.